src/promptflow/promptflow/_utils/exception_utils.py

```diff
@@ -198,55 +198,48 @@ def build_debug_info(self, ex: Exception):
             "innerException": inner_exception,
         }
 
-    def to_dict(self, *, include_debug_info=False):
-        """Return a dict representation of the exception.
+    @property
+    def error_codes(self):
+        """The hierarchy of the error codes.
 
-        This dict specification corresponds to the specification of the Microsoft API Guidelines:
+        We follow the "Microsoft REST API Guidelines" to define error codes in a hierarchy style.
+        See the below link for details:
         https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses
 
-        Note that this dict representation the "error" field in the response body of the API.
-        The whole error response is then populated in another place outside of this class.
+        For general exceptions, the hierarchy should be:
+        ["SystemError", {exception type name}]
```

Comment:
We can remove these two lines from the docstring. Instead, add something like:

This method returns the error codes in a list. It will be converted into a nested json format by `error_code_recursed`.

Reply:
Done.